thoth-export-server/src/xml/doideposit_crossref.rs

```diff
@@ -692,12 +707,8 @@ fn write_chapter_doi_collection<W: Write>(
             ));
         }
     } else {
-        // `doi_data` element is mandatory for `content_item`, and must contain
-        // both `doi` element and `resource` (landing page) element
-        return Err(ThothError::IncompleteMetadataRecord(
-            DEPOSIT_ERROR.to_string(),
-            "Missing chapter DOI".to_string(),
-        ));
+        // Caller should only pass in chapters which have DOIs
+        unreachable!()
```

Comment:
I guess we can now get rid off this condition and simply unwrap the DOI

```
let doi = chapter
        .related_work
        .doi
        .as_ref()
        .expect("Caller should only pass in chapters which have DOIs");
        ```

Reply:
Oh yes, nice :)